Refit/RefitSettings.cs

```diff
@@ -13,21 +12,42 @@ namespace Refit
 {
     public class RefitSettings
     {
+        JsonSerializerSettings jsonSerializerSettings;
+
         public RefitSettings()
         {
             UrlParameterFormatter = new DefaultUrlParameterFormatter();
             FormUrlEncodedParameterFormatter = new DefaultFormUrlEncodedParameterFormatter();
+            ContentSerializer = new JsonContentSerializer();
         }
 
         public Func<Task<string>> AuthorizationHeaderValueGetter { get; set; }
         public Func<HttpMessageHandler> HttpMessageHandlerFactory { get; set; }
 
-        public JsonSerializerSettings JsonSerializerSettings { get; set; }
+        [Obsolete("Set RefitSettings.ContentSerializer = new JsonContentSerializer(JsonSerializerSettings) instead.", false)]
+        public JsonSerializerSettings JsonSerializerSettings
+        {
+            get => jsonSerializerSettings;
+            set
+            {
+                jsonSerializerSettings = value;
+                ContentSerializer = new JsonContentSerializer(value);
+            }
+        }
+
+        public IContentSerializer ContentSerializer { get; set; }
         public IUrlParameterFormatter UrlParameterFormatter { get; set; }
         public IFormUrlEncodedParameterFormatter FormUrlEncodedParameterFormatter { get; set; }
         public bool Buffered { get; set; } = true;
     }
 
+    public interface IContentSerializer
+    {
+        Task<HttpContent> SerializeAsync(object item);
```

Comment:
Can we make this generic, so it's `SerializeAsync<T>(T item)` that could prevent a box and allows for a custom serializer that's optimized for a particular type.

Reply:
That was an easier change than I was expecting! The method swapped over to a generic with no issues.